presto-main/src/main/java/com/facebook/presto/security/FileBasedSystemAccessControl.java

```diff
@@ -0,0 +1,328 @@
+    {
+        if (!canAccessCatalog(identity, table.getCatalogName())) {
+            denyDeleteTable(table.toString(), CATALOG_ACCESS_DENIED + table.getCatalogName());
+        }
+    }
+
+    @Override
+    public void checkCanCreateView(Identity identity, CatalogSchemaTableName view)
+    {
+        if (!canAccessCatalog(identity, view.getCatalogName())) {
+            denyCreateView(view.toString(), CATALOG_ACCESS_DENIED + view.getCatalogName());
+        }
+    }
+
+    @Override
+    public void checkCanDropView(Identity identity, CatalogSchemaTableName view)
+    {
+        if (!canAccessCatalog(identity, view.getCatalogName())) {
+            denyDropView(view.toString(), CATALOG_ACCESS_DENIED + view.getCatalogName());
+        }
+    }
+
+    @Override
+    public void checkCanSelectFromView(Identity identity, CatalogSchemaTableName view)
+    {
+        if (!canAccessCatalog(identity, view.getCatalogName())) {
+            denySelectView(view.toString(), CATALOG_ACCESS_DENIED + view.getCatalogName());
+        }
+    }
+
+    @Override
+    public void checkCanCreateViewWithSelectFromTable(Identity identity, CatalogSchemaTableName table)
+    {
+        if (!canAccessCatalog(identity, table.getCatalogName())) {
+            denyCreateViewWithSelect(table.toString(), CATALOG_ACCESS_DENIED + table.getCatalogName());
+        }
+    }
+
+    @Override
+    public void checkCanCreateViewWithSelectFromView(Identity identity, CatalogSchemaTableName view)
+    {
+        if (!canAccessCatalog(identity, view.getCatalogName())) {
+            denyCreateViewWithSelect(view.toString(), CATALOG_ACCESS_DENIED + view.getCatalogName());
+        }
+    }
+
+    @Override
+    public void checkCanSetCatalogSessionProperty(Identity identity, String catalogName, String propertyName)
+    {
+        if (!canAccessCatalog(identity, catalogName)) {
+            denySetCatalogSessionProperty(catalogName, propertyName, CATALOG_ACCESS_DENIED + catalogName);
+        }
+    }
+
+    @Override
+    public void checkCanGrantTablePrivilege(Identity identity, Privilege privilege, CatalogSchemaTableName table)
+    {
+        if (!canAccessCatalog(identity, table.getCatalogName())) {
+            denyGrantTablePrivilege(privilege.name(), table.toString(), CATALOG_ACCESS_DENIED + table.getCatalogName());
+        }
+    }
+
+    @Override
+    public void checkCanRevokeTablePrivilege(Identity identity, Privilege privilege, CatalogSchemaTableName table)
+    {
+        if (!canAccessCatalog(identity, table.getCatalogName())) {
+            denyRevokeTablePrivilege(privilege.name(), table.toString(), CATALOG_ACCESS_DENIED + table.getCatalogName());
+        }
+    }
+
+    private boolean canAccessCatalog(Identity identity, String catalogName)
+    {
+        for (CatalogAccessControlRule rule : catalogRules) {
+            Optional<Boolean> allowed = rule.match(identity.getUser(), catalogName);
```

Comment:
This doesn't support roles as of now. But we can think about it. Since it's only the hive connector for which there is support for handling roles, I am not sure how useful this feature will be.